relay-server/src/services/buffer/envelope_store/sqlite.rs

```diff
@@ -476,25 +484,82 @@ impl SqliteEnvelopeStore {
     }
 }
 
+fn pack_envelopes(envelopes: Vec<DatabaseEnvelope>) -> Box<[u8]> {
+    let mut packed = vec![];
+    for envelope in envelopes {
+        packed.extend_from_slice(&envelope.received_at.to_le_bytes());
+        packed.extend_from_slice(&(envelope.encoded_envelope.len() as u32).to_le_bytes());
+        packed.extend_from_slice(&envelope.encoded_envelope);
+    }
+    packed.into_boxed_slice()
+}
+
+fn unpack_envelopes(
+    own_key: ProjectKey,
+    sampling_key: ProjectKey,
+    data: &[u8],
+) -> Result<Vec<DatabaseEnvelope>, std::io::Error> {
+    let mut envelopes = vec![];
+    let mut bytes = data.reader();
+    loop {
+        let mut b = [0u8; 8];
+        let Ok(()) = bytes.read_exact(&mut b) else {
```

Comment:
If we have invalid data, we should notify the caller and not break, otherwise, we will never know if we have inconsistent data in the database.

I get why you didn't do it here but it might still be useful to know if something went south.

Reply:
Refactored this now to only break if there's exactly zero remaining bytes.